src/crimson/common/gated.h

```diff
@@ -31,13 +31,13 @@ class Gated {
 
   template <typename Func, typename T>
   inline void dispatch_in_background(const char* what, T& who, Func&& func) {
-    ceph_assert(seastar::this_shard_id() == sid);
+    //ceph_assert(seastar::this_shard_id() == sid);
```

Comment:
Is `io_handler` still related to this violation?

Reply:
@cyx1231st Yes, but I'm not sure if it's the only one, I ran some local tests and found out it failed for io_handler, but it can be in other cases, that's the reason I opened a new tracker and commented those added asserts

Reply:
OK, please let me know if `io_handler` still breaks after https://github.com/ceph/ceph/pull/58986/commits/4f7738685927b081a111c23b10b8ad17b67a533d